source/Meadow.Core/Cloud/HealthReporter.cs

```diff
@@ -88,6 +103,16 @@ public async Task Send()
             ce.Measurements.Add("info.coprocessor_os_version", device.Information.CoprocessorOSVersion);
         }
 
+        foreach (var metric in _customMetrics)
+        {
+            ce.Measurements.TryAdd(metric.Key, metric.Value.Invoke());
```

Comment:
What happens if the app-supplied Action throws?  I think we should protect with a try/catch inside the loop

Reply:
good catch